src/services/server/voter-registration-repository/voter-registration-repository.ts

```diff
@@ -0,0 +1,72 @@
+import 'server-only';
+import { VoterRepository, RegisterBody } from './voter-registration';
+import { ServerError } from '@/errors/server-error';
+import { SERVER_SERVICE_KEYS } from '../keys';
+import { inject } from 'undecorated-di';
+import { Encryptor } from '../encryptor/encryptor';
+import type { CreateSupabaseClient } from '../create-supabase-client/create-supabase-client';
+import { PRIVATE_ENVIRONMENT_VARIABLES } from '@/constants/private-environment-variables';
+
+export const VoterRegistrationRepository = inject(
+  /**
+   * handles voter registration proccess
+   * @example
+   */
+  class SupabaseVoterRepository implements VoterRepository {
+    constructor(
+      private createSupabaseClient: CreateSupabaseClient,
+      private encryptor: Encryptor,
+    ) {}
+    /**
+     * @insertVoterRegistrationInfo
+     * @param RegisterBody - Voter registration information to insert
+     */
+    async insertVoterRegistrationInfo(
+      id: string,
+      RegisterBody: RegisterBody,
+    ): Promise<void> {
+      const dataEncryptor = this.encryptor;
+
+      //helper function that encrypts the register body
+      const encryptRegisterBody = async (
+        obj: typeof RegisterBody,
+      ): Promise<typeof RegisterBody> => {
+        const cryptoKey = await PRIVATE_ENVIRONMENT_VARIABLES.CRYPTO_KEY;
+
+        const encryptedObject = { ...obj };
+        for (const key in encryptedObject) {
+          if (Object.prototype.hasOwnProperty.call(encryptedObject, key)) {
```

Comment:
Maybe just do:
```
for (const key of Object.keys(encryptedObject)) {
```

Object.keys won't return inherited keys, making the `Object.prototype.hasOwnProperty.call` check unnecessary. I'm also a fan of using Object.entries when I need both the key and value:

```
for (const [key, value] of Object.entries(encryptedObject)) {
```